src/core/thread/mle.cpp

```diff
@@ -1940,6 +1947,19 @@ void Mle::HandleMessageTransmissionTimer(void)
     //  - Retransmission of "Child Update Request",
     //  - Retransmission of "Data Request" on a child,
     //  - Sending periodic keep-alive "Child Update Request" messages on a non-sleepy (rx-on) child.
+    //  - Retransmission of "Link Request" after router reset
+
+#if OPENTHREAD_FTD
+    // Retransmit multicast link request if no response has been received
+    // and maximum transmission limit has not been reached.
+    if (mInRouterResetSync && mLinkRequestAttempts < mMaxLinkRequests)
+    {
+        IgnoreError(Get<MleRouter>().SendLinkRequest(nullptr));
```

Comment:
`SendLinkRequest(nullptr)` has this check at the top:
- `VerifyOrExit(mLinkRequestDelay == 0 && mChallengeTimeout == 0);`
- So if `mChallengeTimeout != 0` then we won't send.

Do we know it would be zero at this point (when we try to retx Link Req)?

----

I did check the numbers and code around `mChallengeTimeout`:

- We set `mChallengeTimeout` only on `SendLinkRequest(nullptr)` 
- We set it as `mChallengeTimeout = (((2 * kMaxResponseDelay) + kStateUpdatePeriod - 1) / kStateUpdatePeriod);`
    - Both `kMaxResponseDelay` and `kStateUpdatePeriod` are `1000`, so `2999/1000`
     - Fancy way of setting it to `2` :)  
- And we decrement it from `HandleTimeTick()` (so roughly 2 seconds).
- The new mechanism for retx of Link Req uses random `[4.5-5.5]` sec, so we should be good.